### PR TITLE
new resource: servicecatalog_order

### PR DESCRIPTION
**Pull request by marcosdiez**
A Service Now ServiceCatalog order. It consumes the "order_now" API ( https://developer.servicenow.com/dev.do#!/reference/api/utah/rest/c_ServiceCatalogAPI#servicecat-POST-items-order_now )

Please note that Service Now catalog orders are immutable.
That means if you try to destroy an order, it will be removed from terraform but nothing will happen in Service Now.
Also, if you try to update an order, it will be removed from terraform and a new one will be created. In other words, you will end up with two orders in service now.

### CODE REVIEW

**Comment by Andrei-Predoiu** [comment]
Hi @marcosdiez How will this resource be used irl? 
From what i can imagine it sounds like something like this, is that accurate?

You want to create orders for SN when something is done in TF so that humans can take action?

**Comment by marcosdiez** [comment]
Hello @Andrei-Predoiu !

Good question.

Where I work, every time we use some resources we have to request permission to the team that owns them.
And that's done via service now.

So we'll have two use cases:

* Create a terraform module that will not only create the resource we need but also officially request it on service now so the team that owns it can track it's usage. 
* Create a terraform module that requests the item in service now, checks if it's approved or not, fail before the approval, when run again and validate that it's a approved it will continue and create the resource.

This will at very least guarantee consistency in the service now request and the object itself that are created with terraform.

Neither are my dream workflow, but both approaches will allow my department to abstract service now request creation from our lives while following all the guidelines from the security department and our auditors.

File: internal/models/servicecatalog_order.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		Computed:    true,
+	},
+	"number": {
+		Description: "Number of the generated request",
+		Type:        schema.TypeString,
+		Computed:    true,
+	},
+	// this field is not documented, causes errors and is not needed
+	// "$$uiNotification": {
+	// 	Type:     schema.TypeSet,
```
Comment (Andrei-Predoiu) [inline internal/models/servicecatalog_order.go:79]: This should be cleaned up

**Review by Andrei-Predoiu — CHANGES_REQUESTED** [review]

**Comment by marcosdiez** [comment]
@Andrei-Predoiu done!

**Review by Andrei-Predoiu — APPROVED** [review]

**Pull request merged**